gadget/update.go

```diff
@@ -426,7 +426,7 @@ func EnsureVolumeCompatibility(gadgetVolume *Volume, diskVolume *OnDiskVolume, o
 			// not touch it, unless a gadget asset update says to update that
 			// image file with a new binary image file. This also covers the
 			// partial filesystem case.
-			if gs.Filesystem != "" && gs.Filesystem != ds.PartitionFSType {
+			if gs.Filesystem != "" && !gs.MatchesLinuxFilesystem(ds.PartitionFSType) {
```

Comment:
do we need to do the reverse of MatchesLinuxFilesystem in:

./install/partition.go:229:			PartitionFSType:  vs.Filesystem,
./layout.go:285:	los.PartitionFSType = los.Filesystem()

these are the ones I found, don't know if there is something more indirect somewhere

Reply:
That's a good point. I've now created an ancillary method to do this reverse conversion, as we need it here and also in some other cases like when mounting partitions.